python/paddle/fluid/regularizer.py

```diff
@@ -231,6 +263,95 @@ def __str__(self):
         return "L1Decay, regularization_coeff=%f" % self._regularization_coeff
 
 
+class WeightDecayBase(object):
+    def __str__(self):
+        raise NotImplementedError()
+
+    def __call__(self, params_and_grads):
```

Comment:
The comments should be in the base class rather than all implementation.

All objective oriented programming should depend on the interface and be interface oriented.